Error in grouping panel when toggling 'Show duplicates only'

#### System information: 
- Operating system (distribution) and version: Windows 10
- DBeaver version: 22.1.0

#### Connection specification:
- Database name and version: Oracle Database 19c
- Driver name: Ojdbc8 version 19.3.0.0
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:

When activating the 'Show duplicates only (COUNT > 1)' toggle button, it breakes because the alias given to the Count(*) function (new in DBeaver 22, it didn't happen with DBeaver 21) is used in the HAVING clause, which does not admit select aliases.

#### Steps to reproduce, if exist:

1. Execute a query that returns some rows, and open the Grouping panel.
2. Add a column to the Grouping.
3. Press the 'Show duplicates only (COUNT > 1)' toggle button.

#### Include any warning/errors/backtraces from the logs
Query executed:
`SELECT 1 id, 'des' description FROM dual`


Query shown:
```
SELECT ID, COUNT(*) as count_ FROM (
SELECT 1 id, 'des' description FROM dual
) src
GROUP BY ID
HAVING count_ > 1
```
Error message:
`SQL Error [904] [42000]: ORA-00904: "COUNT_": invalid identifier`
<details>
  <summary>Stacktrace</summary>
  
org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [904] [42000]: ORA-00904: "COUNT_": invalid identifier

	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:133)
	at org.jkiss.dbeaver.ui.controls.resultset.panel.grouping.GroupingDataContainer.readData(GroupingDataContainer.java:109)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:118)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:172)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:116)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:4868)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

Caused by: java.sql.SQLSyntaxErrorException: ORA-00904: "COUNT_": invalid identifier

	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:509)
	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:461)
	at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1104)
	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:550)
	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:268)
	at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:655)
	at oracle.jdbc.driver.T4CStatement.doOall8(T4CStatement.java:229)
	at oracle.jdbc.driver.T4CStatement.doOall8(T4CStatement.java:41)
	at oracle.jdbc.driver.T4CStatement.executeForDescribe(T4CStatement.java:765)
	at oracle.jdbc.driver.OracleStatement.executeMaybeDescribe(OracleStatement.java:983)
	at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1168)
	at oracle.jdbc.driver.OracleStatement.executeInternal(OracleStatement.java:1823)
	at oracle.jdbc.driver.OracleStatement.execute(OracleStatement.java:1778)
	at oracle.jdbc.driver.OracleStatementWrapper.execute(OracleStatementWrapper.java:303)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:329)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.lambda$0(JDBCStatementImpl.java:131)
	at org.jkiss.dbeaver.utils.SecurityManagerUtils.wrapDriverActions(SecurityManagerUtils.java:94)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	... 7 more

Caused by: Error : 904, Position : 103, Sql = SELECT ID, COUNT(\*) as count_ FROM (
SELECT 1 id, 'des' description FROM dual
) src
GROUP BY ID
HAVING count_ > 1, OriginalSql = SELECT ID, COUNT(\*) as count_ FROM (
SELECT 1 id, 'des' description FROM dual
) src
GROUP BY ID
HAVING count_ > 1, Error Msg = ORA-00904: "COUNT_": invalid identifier


	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:513)
	... 24 more
</details>

--- a/plugins/org.jkiss.dbeaver.ext.db2.i/src/org/jkiss/dbeaver/ext/db2/i/model/DB2ISQLDialect.java
+++ b/plugins/org.jkiss.dbeaver.ext.db2.i/src/org/jkiss/dbeaver/ext/db2/i/model/DB2ISQLDialect.java
@@ -48,4 +48,9 @@ public class DB2ISQLDialect extends GenericSQLDialect {
     public MultiValueInsertMode getDefaultMultiValueInsertMode() {
         return MultiValueInsertMode.GROUP_ROWS;
     }
+
+    @Override
+    public boolean supportsAliasInConditions() {
+        return false;
+    }
 }

--- a/plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/DB2SQLDialect.java
+++ b/plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/DB2SQLDialect.java
@@ -163,4 +163,9 @@ public class DB2SQLDialect extends JDBCSQLDialect implements TPRuleProvider {
             rules.add(new SQLMultiWordRule(new String[]{"ROW", "END"}, keywordToken));
         }
     }
+
+    @Override
+    public boolean supportsAliasInConditions() {
+        return false;
+    }
 }

--- a/plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/ExasolSQLDialect.java
+++ b/plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/ExasolSQLDialect.java
@@ -203,5 +203,10 @@ public class ExasolSQLDialect extends JDBCSQLDialect {
         return true;
     }
 
+    @Override
+    public boolean supportsAliasInConditions() {
+        return false;
+    }
+
 }
 

--- a/plugins/org.jkiss.dbeaver.ext.firebird/src/org/jkiss/dbeaver/ext/firebird/model/FireBirdSQLDialect.java
+++ b/plugins/org.jkiss.dbeaver.ext.firebird/src/org/jkiss/dbeaver/ext/firebird/model/FireBirdSQLDialect.java
@@ -95,4 +95,9 @@ public class FireBirdSQLDialect extends GenericSQLDialect {
     public boolean supportsInsertAllDefaultValuesStatement() {
         return true;
     }
+
+    @Override
+    public boolean supportsAliasInConditions() {
+        return false;
+    }
 }

--- a/plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDialect.java
+++ b/plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDialect.java
@@ -353,4 +353,9 @@ public class SQLServerDialect extends JDBCSQLDialect implements TPRuleProvider {
     public boolean supportsInsertAllDefaultValuesStatement() {
         return isSqlServer; // Sybase throws a syntax error on "DEFAULT" keyword
     }
+
+    @Override
+    public boolean supportsAliasInConditions() {
+        return false;
+    }
 }

--- a/plugins/org.jkiss.dbeaver.ext.oracle/src/org/jkiss/dbeaver/ext/oracle/model/OracleSQLDialect.java
+++ b/plugins/org.jkiss.dbeaver.ext.oracle/src/org/jkiss/dbeaver/ext/oracle/model/OracleSQLDialect.java
@@ -658,4 +658,9 @@ public class OracleSQLDialect extends JDBCSQLDialect implements SQLDataTypeConve
     public boolean hasCaseSensitiveFiltration() {
         return true;
     }
+
+    @Override
+    public boolean supportsAliasInConditions() {
+        return false;
+    }
 }

